### PR TITLE
RDKB-57439 : [XER10-UK] IPv6 Handling - Wan & Lan Management.

### PR DESCRIPTION
**Pull request by LakshminarayananShenbagaraj**
Reason for change:
IPv6 Handling Requirement for Sky Products are different from Comcast. Sky using IAPD , DHCPv6 prefix delegation to provide prefixes for use on the subscriber LAN as well as to get WAN IPv6 address from backend.

The implementation for this feature is under CcspPandM components so that needs to be enable for applicable products like Sky. So we need to control RDKB_GLOBAL_PRODUCT_REQ common MACRO with feature applicable flags with respect to region.

Test Procedure:
1. IPv6 ULA should work for Sky but those params writable permission should be prohibited for comcast products.
2. WAN and LAN functionality should work without any issue.

Risks: Medium

Signed-off-by: Lakshminarayanan <lakshminarayanan.shenbagaraj2@sky.uk>

### CODE REVIEW

File: source/WanManager/wanmgr_dhcpv6_apis.c [hunk 1]
```diff
@@ -1484,12 +1484,17 @@ int dhcpv6_assign_global_ip(char * prefix, char * intfName, char * ipAddr)
         CcspTraceError(("error, there is not '::' in prefix:%s\n", prefix));
         return 1;
     }
-#ifdef _HUB4_PRODUCT_REQ_
-    if(strncmp(intfName, COSA_DML_DHCPV6_SERVER_IFNAME, strlen(intfName)) == 0)
+#if defined(_HUB4_PRODUCT_REQ_) || defined(_RDKB_GLOBAL_PRODUCT_REQ_)
+#if defined(_RDKB_GLOBAL_PRODUCT_REQ_)
+    if ( TRUE == WanMgr_Util_IsFeatureApplicable(SYSEVENT_FEATURE_DISABLE_IPV6_EUI64_FORMAT_SUPPORT, INPUT_SOURCE_TYPE_SYSEVENT) )
```
Comment (guto86) [inline source/WanManager/wanmgr_dhcpv6_apis.c:1489]: Would be possible to use a configuration in PSM instead of depending of a sysevent?

**Review by guto86 — CHANGES_REQUESTED** [review]

File: source/WanManager/wanmgr_dhcpv6_apis.c (hunk 1 above)
Comment (LakshminarayananShenbagaraj) [reply source/WanManager/wanmgr_dhcpv6_apis.c:1489]: Can we introduce this PSM member when we are unifying this? If we introduce new PSM then we need to consider image upgrade use case for existing products so all the default configuration member needs to be update. As per the current PSM migration code, protected under GLOBAL product requirement flag only not for other products.

Please Advise.

Comment (LakshminarayananShenbagaraj) [reply source/WanManager/wanmgr_dhcpv6_apis.c]: Addressed all your comments as part of latest update. Could you please check?

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c [hunk 2]
```diff
@@ -1560,6 +1560,42 @@ ANSC_STATUS WanMgr_WanConfInit (DML_WANMGR_CONFIG* pWanConfig)
...
     pWanConfig->RestorationDelay = wan_restoration_delay;
 
+    memset(param_name, 0, sizeof(param_name));
+    memset(param_value, 0, sizeof(param_value));
+    _ansc_sprintf(param_name, PSM_WANMANAGER_BACKUPWANDNS_SUPPORT);
+    retPsmGet = WanMgr_RdkBus_GetParamValuesFromDB(param_name,param_value,sizeof(param_value));
+    if ((retPsmGet == CCSP_SUCCESS) && (param_value[0] != '\0') && (0 == strncmp(param_value, "TRUE", 4)))
+        pWanConfig->BackupWanDnsSupport = TRUE;
+    else
+        pWanConfig->BackupWanDnsSupport = FALSE;
```
Comment (guto86) [inline source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c:1570]: BackupWanDnsSupport consider the default as true

File: source/WanManager/wanmgr_data.c [hunk 3]
```diff
@@ -69,6 +69,10 @@ void WanMgr_SetConfigData_Default(DML_WANMGR_CONFIG* pWanDmlConfig)
         pWanDmlConfig->DeviceNwMode = GATEWAY_MODE;
         pWanDmlConfig->DeviceNwModeChanged = FALSE;
         pWanDmlConfig->BootToWanUp = FALSE;
+        pWanDmlConfig->BackupWanDnsSupport = FALSE;
```
Comment (guto86) [inline source/WanManager/wanmgr_data.c:72]: default is true

File: source/WanManager/wanmgr_dhcpv6_apis.c [hunk 4]
```diff
@@ -1484,12 +1484,26 @@ int dhcpv6_assign_global_ip(char * prefix, char * intfName, char * ipAddr)
         CcspTraceError(("error, there is not '::' in prefix:%s\n", prefix));
         return 1;
     }
-#ifdef _HUB4_PRODUCT_REQ_
-    if(strncmp(intfName, COSA_DML_DHCPV6_SERVER_IFNAME, strlen(intfName)) == 0)
+#if defined(_HUB4_PRODUCT_REQ_) || defined(_RDKB_GLOBAL_PRODUCT_REQ_)
+#if defined(_RDKB_GLOBAL_PRODUCT_REQ_)
+    WanMgr_Config_Data_t    *pWanConfigData = WanMgr_GetConfigData_locked();
+    unsigned char           IPv6EUI64FormatSupport = FALSE;
```
Comment (guto86) [inline source/WanManager/wanmgr_dhcpv6_apis.c:1490]: default is true

File: source/WanManager/wanmgr_interface_sm.c [hunk 5]
```diff
@@ -1255,19 +1255,35 @@ static int wan_tearDownIPv4(WanMgr_IfaceSM_Controller_t * pWanIfaceCtrl)
     DML_VIRTUAL_IFACE* p_VirtIf = WanMgr_getVirtualIfaceById(pInterface->VirtIfList, pWanIfaceCtrl->VirIfIdx);
 
     /** Reset IPv4 DNS configuration. */
-#if (defined (_XB6_PRODUCT_REQ_) || defined (_CBR2_PRODUCT_REQ_) || defined(_PLATFORM_RASPBERRYPI_)) 
+#if (defined (_XB6_PRODUCT_REQ_) || defined (_CBR2_PRODUCT_REQ_) || defined(_PLATFORM_RASPBERRYPI_) || defined (_RDKB_GLOBAL_PRODUCT_REQ_))
+#if defined (_RDKB_GLOBAL_PRODUCT_REQ_)
+    WanMgr_Config_Data_t    *pWanConfigData = WanMgr_GetConfigData_locked();
+    unsigned char           BackupWanDnsSupport = FALSE;
```
Comment (guto86) [inline source/WanManager/wanmgr_interface_sm.c:1261]: default is true

**Review by guto86 — CHANGES_REQUESTED** [review]

**Review by guto86 — APPROVED** [review]

**Pull request merged**